components/tabs/src/sync/store.rs

```diff
@@ -70,6 +89,12 @@ impl<'a> Store for TabsStore<'a> {
         "tabs"
     }
 
+    fn prepare_for_sync(&self, engine: &clients::Engine<'_>) -> Result<(), failure::Error> {
+        self.remote_clients
+            .replace(engine.recent_client_list.clone());
```

Comment:
Unfortunately, I don't think there's a way to avoid this `clone`. The store outlives the clients engine, so, even though we "know" we'll only touch the recent client list during a sync, there's no way to tell the compiler that. Additionally, all the stores get created before we call `sync_multiple`, while the clients engine gets created inside that function, so we can't tie the store's lifetime to clients.

This `clone` is still much, much less expensive than making a network request to fetch the clients collection again, though. The advantage of fetching clients again is that we could use this without the sync manager, but given that 1) we only use the collection for more accurate device names and types, and 2) we're encouraging everyone to use the sync manager, instead of one-off syncs, this is fine.